inc/front/async-css.php

```diff
@@ -105,10 +105,28 @@ function rocket_insert_critical_css() {
 		return;
 	}
 
-	$critical_css = wp_kses( get_rocket_option( 'critical_css' ), array( '\'', '\"' ) );
-	$critical_css = str_replace( '&gt;', '>', $critical_css );
-
-	echo '<style id="rocket-critical-css">' . $critical_css . '</style>';
+	$critical_css = get_rocket_option( 'critical_css' );
+
+	if ( is_home() ) {
+		$critical_css_content = $critical_css['home'];
+	} else if ( is_category() ) {
+		$critical_css_content = $critical_css['category'];
+	} else if ( is_tag() ) {
+		$critical_css_content = $critical_css['post_tag'];
+	} else if ( is_tax() ) {
+		$taxonomy = get_queried_object()->term_name;
+		$critical_css_content = $critical_css[ $taxonomy ];
+	} else if ( is_singular() ) {
+		$post_type = get_post_type();
+		$critical_css_content = $critical_css[ $post_type ];
+	} else {
+		$critical_css_content = $critical_css['front_page'];
+	}
+	
+	$critical_css_content = wp_kses( $critical_css_content, array( '\'', '\"' ) );
```

Comment:
2nd argument of `wp_kses()` must be a "[List of allowed HTML elements](https://wpseek.com/function/wp_kses/)". Am I missing something?

Reply:
This was done like this because of https://github.com/wp-media/wp-rocket/commit/a3d92df0b2f61724c00a98f2bae6e1ddb72edc66#diff-edd9f7b76d6d976f0d41fec8cbfc1812 but there might be a better way